test/cri-containerd/createcontainer_test.go

```diff
@@ -793,3 +795,74 @@ func Test_CreateContainer_CPUShares_LCOW(t *testing.T) {
 	}
 	runCreateContainerTest(t, lcowRuntimeHandler, request)
 }
+
+func Test_CreateContainer_File_Hostpath_LCOW(t *testing.T) {
+	pullRequiredLcowImages(t, []string{imageLcowK8sPause, imageLcowAlpine})
+
+	tempFile, err := ioutil.TempFile("", "test")
+
+	defer tempFile.Close()
```

Comment:
You don't need to defer this. You can just call `Close()`. But for your education, on Windows this order doesn't work. On Linux this is a natural thing to write but for Windows you are calling `Remove` before `Close` which means the `Remove` fails with an error.

Reply:
Also, only ever defer AFTER a check to `err != nil` many times the pattern is `return nil, err` so this is likely to panic